ev3dev2/motor.py

```diff
@@ -2237,30 +2224,103 @@ def follow_gyro_angle(self,
 
         self.stop()
 
-    def turn_to_angle_gyro(self,
+    def _turn(
+            self,
             speed,
-            target_angle=0,
+            target_angle,
+            brake=True,
             wiggle_room=2,
             sleep_time=0.01
         ):
         """
-        Pivot Turn
+        Use a GyroSensor to rotate in place for ``target_angle``
 
         ``speed`` is the desired speed of the midpoint of the robot
 
-        ``target_angle`` is the target angle we want to pivot to
+        ``target_angle`` is the number of degrees we want to rotate
+
+        ``brake`` hit the brakes once we reach ``target_angle``
 
         ``wiggle_room`` is the +/- angle threshold to control how accurate the turn should be
 
         ``sleep_time`` is how many seconds we sleep on each pass through
             the loop.  This is to give the robot a chance to react
             to the new motor settings. This should be something small such
             as 0.01 (10ms).
+        """
+
+        # MoveTank does not have information on wheel size and distance (that is
+        # MoveDifferential) so we must use a GyroSensor to control how far we rotate.
+        assert self._gyro, "GyroSensor must be defined"
+
+        if not target_angle:
+            return
+
+        speed_native_units = speed.to_native_units(self.left_motor)
+        init_angle = self._gyro.angle
+        first = True
+
+        while True:
+            current_angle = self._gyro.angle
+            degrees_moved = abs(current_angle - init_angle)
+            delta = abs(abs(target_angle) - degrees_moved)
+
+            if delta <= wiggle_room:
+                self.stop(brake=brake)
+                break
+
+            # our goal is to rotate target_angle clockwise
+            if target_angle > 0:
+
+                # we have not gone far enough, rotate clockwise
+                if first or degrees_moved < target_angle:
```

Comment:
The `first` is needed so that on our first pass through this loop we start the motors in the correct direction...although now that I type that and re-read this code it doesn't seem like it would be needed.  Will test with this removed.

Reply:
`If we're too far to the left, turn right; otherwise, turn left or similar` it depends on if our target_angle is positive or negative...take a look at the speeds that are being set in the various blocks of this if/else chain.

Reply:
confirmed that `first` is not needed

Reply:
`If we're too far to the left, turn right; otherwise, turn left or similar` you are correct this code path can be simplified...have chopped a good bit here